### ISSUE TITLE
data shape

### ISSUE THREAD

**Issue by piemmea** [LeiCao97/PreLnc#3]
Hi, 
I'm (trying to) using PreLnc but I get the following error when calling sklearn:

> ValueError: Expected 2D array, got 1D array instead

do you know how to overcome this problem? I am using pyton 2.7, Biophyton 1.76, sklearn 0.20
tnx!

**Comment by etlioglu** [comment]
TLDR: This happens when a single sequence is supplied with the `-i` parameter. Just add another fasta entry to your input file and neglect the second line of the output.

I have experienced the same issue and realized that this happens when using fasta files containing just one sequence. `PreLnc/example/example.fa` fails with the same message, output below.

I believe this has something to do with some changes in `numpy` that `PreLnc` relies on. At this point I am not planning to submit a PR because I did not see any unit tests associated with the package and I don't want to break something else but also because of the fact that there is a simple work around. One can add a second sequence and neglect the second line of the output.

```
.../py2env/lib/python2.7/site-packages/Bio/Seq.py:182: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.
  BiopythonWarning,

[PreLnc]start to predict
.../py2env/lib/python2.7/site-packages/sklearn/base.py:251: UserWarning: Trying to unpickle estimator DecisionTreeClassifier from version 0.20.3 when using version 0.20.0. This might lead to breaking code or invalid results. Use at your own risk.
  UserWarning)
.../py2env/lib/python2.7/site-packages/sklearn/base.py:251: UserWarning: Trying to unpickle estimator RandomForestClassifier from version 0.20.3 when using version 0.20.0. This might lead to breaking code or invalid results. Use at your own risk.
  UserWarning)
Traceback (most recent call last):
  File "../bin/prelnc.py", line 412, in <module>
    main()
  File "../bin/prelnc.py", line 397, in main
    predict_result= load_model.predict_proba(data)[:,1]
  File ".../py2env/lib/python2.7/site-packages/sklearn/ensemble/forest.py", line 585, in predict_proba
    X = self._validate_X_predict(X)
  File ".../py2env/lib/python2.7/site-packages/sklearn/ensemble/forest.py", line 364, in _validate_X_predict
    return self.estimators_[0]._validate_X_predict(X, check_input=True)
  File ".../py2env/lib/python2.7/site-packages/sklearn/tree/tree.py", line 376, in _validate_X_predict
    X = check_array(X, dtype=DTYPE, accept_sparse="csr")
  File ".../py2env/lib/python2.7/site-packages/sklearn/utils/validation.py", line 547, in check_array
    "if it contains a single sample.".format(array))
ValueError: Expected 2D array, got 1D array instead:
array=[ 1.1000000e-01  3.8643000e-01 -1.2057361e-01  2.7200000e+02
  5.5250001e-01  1.2155000e+00  4.0349998e+01  2.1602471e+00
  5.6045532e+00  2.2100000e-01  1.3260000e+00  9.0700000e+02
  2.2100000e-01  1.5470001e+00  6.6299999e-01  1.1050000e+00
  2.0994000e+00  1.7680000e+00  1.0000000e+00  2.0994000e+00].
Reshape your data either using array.reshape(-1, 1) if your data has a single feature or array.reshape(1, -1) if it contains a single sample.
```
